packages/vc-verification/src/issuer-verification-offchainclaims.ts

```diff
@@ -0,0 +1,134 @@
+import * as jwt from 'jsonwebtoken';
+import { providers } from 'ethers';
+import { ProofVerifier } from '@ew-did-registry/claims';
+import { Resolver, addressOf } from '@ew-did-registry/did-ethr-resolver';
+import { RegistrySettings } from '@ew-did-registry/did-resolver-interface';
+import { CredentialResolver, IssuerDefinitionResolver } from '.';
+import { OffChainClaim } from './models';
+
+export class ClaimIssuerVerification {
+  private _resolver: Resolver;
+  private _issuerDefResolver: IssuerDefinitionResolver;
+  private _credentialResolver: CredentialResolver;
+
+  /**
+   *
+   * @param provider
+   * @param registrySetting
+   * @param credentialResolver
+   * @param issuerDefResolver
+   */
+  constructor(
+    provider: providers.Provider,
+    registrySetting: RegistrySettings,
+    credentialResolver: CredentialResolver,
+    issuerDefResolver: IssuerDefinitionResolver
+  ) {
+    this._issuerDefResolver = issuerDefResolver;
+    this._resolver = new Resolver(provider, registrySetting);
+    this._credentialResolver = credentialResolver;
+  }
+
+  /**
+   * Verifies chain of trust for a given holder's DID and role
+   * @param {string} credential
+   */
+  async verifyChainOfTrustClaims(issuerDID: string, role: string) {
+    let currentIssuerDID = issuerDID;
+    while (true) {
+      if (!(await this.verifyIssuerAuthority(role, currentIssuerDID))) {
+        throw new Error('Issuer is not allowed to issue credential');
+      }
+      const roleIssuers = await this._issuerDefResolver.getIssuerDefinition(
+        role
+      );
+      if (await this.isRoleIssuerDID(role)) {
+        return true;
+      } else if (roleIssuers && roleIssuers.roleName) {
+        const offChainClaim = await this._credentialResolver.getOffChainClaim(
```

Comment:
IMO, it's better to return the `issuedToken` from `getOffChainClaim`, IMO. This will avoid the `offChainClaim.issuedToken` references on line 56 and 58 (which reduces the knowledge this method needs on `offChainClaim`'s internals) and could potential remove the need for the additional `if` on line 56
```suggestion
        const issuedToken = await this._credentialResolver.getOffChainClaim(
```